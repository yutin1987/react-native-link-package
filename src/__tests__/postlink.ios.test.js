jest.mock('inquirer');

const _ = require('lodash');
const glob = require('glob');
const inquirer = require('inquirer');
const postlink = require('../postlink.ios');

const option = { ignore: ['node_modules/**', '**/build/**'], realpath: true };

describe('postlink ios', () => {
  it('mount framework', () => (
    postlink(glob.sync('**/*.pbxproj', option)[0], {
      packageName: 'rn-package',
      framework: {
        path: 'ios/Frameworksa/',
        files: ['appkey.framework'],
      },
    })
    .then((result) => {
      expect(result.pbxproj).toContain('../../node_modules/rn-package/ios/Frameworksa/appkey.framework');
      expect(result.pbxproj).toContain('appkey.framework in Frameworks');
      expect(result.pbxproj).toContain('\\"../../node_modules/rn-package/ios/Frameworksa\\"');
    })
  ));

  it('mount params', () => {
    const value = '594f5212-3d66-45d9-a992-ad94263cf318';
    inquirer.prompt.mockReturnValueOnce(Promise.resolve({ value }));

    return postlink(glob.sync('**/*.pbxproj', option)[0], {
      packageName: 'rn-package',
      params: [{
        name: 'APP_KEY',
        message: 'What\'s your app key ?',
      }],
    })
    .then((result) => {
      expect(result.plist).toMatchSnapshot();
      expect(result.plist).toContain('APP_KEY');
      expect(result.plist).toContain(value);
    });
  });

  it('mount params when setup handler', () => {
    const value = '594f5212-3d66-45d9-a992-ad94263cf318';
    inquirer.prompt.mockReturnValueOnce(Promise.resolve({ value }));

    return postlink(glob.sync('**/*.pbxproj', option)[0], {
      packageName: 'rn-package',
      params: [{
        name: 'APP_KEY',
        message: 'What\'s your app key ?',
        handler: (plist, answer) => {
          const CFBundleURLSchemes = _.get(plist, 'CFBundleURLTypes.CFBundleURLSchemes', []);
          CFBundleURLSchemes.push(`ak${answer.value || 'APP_KEY in here'}`);
          _.set(plist, 'CFBundleURLTypes.CFBundleURLSchemes', CFBundleURLSchemes);
        },
      }],
    })
    .then((result) => {
      expect(result.plist).toMatchSnapshot();
      expect(result.plist).toContain('CFBundleURLSchemes');
      expect(result.plist).toContain(`ak${value}`);
    });
  });
});
