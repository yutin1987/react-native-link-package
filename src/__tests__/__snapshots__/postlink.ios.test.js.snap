exports[`postlink ios mount params 1`] = `
"<?xml version=\"1.0\" encoding=\"UTF-8\"?>
<!DOCTYPE plist PUBLIC \"-//Apple//DTD PLIST 1.0//EN\" \"http://www.apple.com/DTDs/PropertyList-1.0.dtd\">
<plist version=\"1.0\">
  <dict>
    <key>CFBundleDevelopmentRegion</key>
    <string>en</string>
    <key>CFBundleExecutable</key>
    <string>$(EXECUTABLE_NAME)</string>
    <key>CFBundleIdentifier</key>
    <string>org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)</string>
    <key>CFBundleInfoDictionaryVersion</key>
    <string>6.0</string>
    <key>CFBundleName</key>
    <string>$(PRODUCT_NAME)</string>
    <key>CFBundlePackageType</key>
    <string>APPL</string>
    <key>CFBundleShortVersionString</key>
    <string>1.0</string>
    <key>CFBundleSignature</key>
    <string>????</string>
    <key>CFBundleVersion</key>
    <string>1</string>
    <key>LSRequiresIPhoneOS</key>
    <true/>
    <key>UILaunchStoryboardName</key>
    <string>LaunchScreen</string>
    <key>UIRequiredDeviceCapabilities</key>
    <array>
      <string>armv7</string>
    </array>
    <key>UISupportedInterfaceOrientations</key>
    <array>
      <string>UIInterfaceOrientationPortrait</string>
      <string>UIInterfaceOrientationLandscapeLeft</string>
      <string>UIInterfaceOrientationLandscapeRight</string>
    </array>
    <key>UIViewControllerBasedStatusBarAppearance</key>
    <false/>
    <key>NSLocationWhenInUseUsageDescription</key>
    <key>NSAppTransportSecurity</key>
    <dict>
      <key>NSExceptionDomains</key>
      <dict>
        <key>localhost</key>
        <dict>
          <key>NSExceptionAllowsInsecureHTTPLoads</key>
          <true/>
        </dict>
      </dict>
    </dict>
    <key>APP_KEY</key>
    <string>594f5212-3d66-45d9-a992-ad94263cf318</string>
  </dict>
</plist>"
`;

exports[`postlink ios mount params when setup handler 1`] = `
"<?xml version=\"1.0\" encoding=\"UTF-8\"?>
<!DOCTYPE plist PUBLIC \"-//Apple//DTD PLIST 1.0//EN\" \"http://www.apple.com/DTDs/PropertyList-1.0.dtd\">
<plist version=\"1.0\">
  <dict>
    <key>CFBundleDevelopmentRegion</key>
    <string>en</string>
    <key>CFBundleExecutable</key>
    <string>$(EXECUTABLE_NAME)</string>
    <key>CFBundleIdentifier</key>
    <string>org.reactjs.native.example.$(PRODUCT_NAME:rfc1034identifier)</string>
    <key>CFBundleInfoDictionaryVersion</key>
    <string>6.0</string>
    <key>CFBundleName</key>
    <string>$(PRODUCT_NAME)</string>
    <key>CFBundlePackageType</key>
    <string>APPL</string>
    <key>CFBundleShortVersionString</key>
    <string>1.0</string>
    <key>CFBundleSignature</key>
    <string>????</string>
    <key>CFBundleVersion</key>
    <string>1</string>
    <key>LSRequiresIPhoneOS</key>
    <true/>
    <key>UILaunchStoryboardName</key>
    <string>LaunchScreen</string>
    <key>UIRequiredDeviceCapabilities</key>
    <array>
      <string>armv7</string>
    </array>
    <key>UISupportedInterfaceOrientations</key>
    <array>
      <string>UIInterfaceOrientationPortrait</string>
      <string>UIInterfaceOrientationLandscapeLeft</string>
      <string>UIInterfaceOrientationLandscapeRight</string>
    </array>
    <key>UIViewControllerBasedStatusBarAppearance</key>
    <false/>
    <key>NSLocationWhenInUseUsageDescription</key>
    <key>NSAppTransportSecurity</key>
    <dict>
      <key>NSExceptionDomains</key>
      <dict>
        <key>localhost</key>
        <dict>
          <key>NSExceptionAllowsInsecureHTTPLoads</key>
          <true/>
        </dict>
      </dict>
    </dict>
    <key>CFBundleURLTypes</key>
    <dict>
      <key>CFBundleURLSchemes</key>
      <array>
        <string>ak594f5212-3d66-45d9-a992-ad94263cf318</string>
      </array>
    </dict>
  </dict>
</plist>"
`;
